distributed/client.py

```diff
@@ -1163,22 +1163,20 @@ async def _start(self, timeout=no_default, **kwargs):
         elif self._start_arg is None:
             from distributed.deploy import LocalCluster
 
+            local_cluster = partial(
+                LocalCluster,
+                loop=self.loop,
+                asynchronous=self._asynchronous,
+                **self._startup_kwargs,
+            )
+
             try:
-                self.cluster = await LocalCluster(
-                    loop=self.loop,
-                    asynchronous=self._asynchronous,
-                    **self._startup_kwargs,
-                )
+                self.cluster = await local_cluster()
             except OSError as e:
                 if e.errno != errno.EADDRINUSE:
```

Comment:
in writing a test for this I found this `except ` clause is unreachable:

LocalCluster(scheduler_port) now defaults to 0
https://github.com/dask/distributed/blob/5feb17151cdf660a3443abf8596444a9f51dc575/distributed/deploy/local.py#L108

If I pass an in-use port to LocalCluster:

```python
def test_client_cluster_synchronous(loop):
    with socket.create_server(("localhost", 0)) as server:
        in_use_port = server.getsockname()[1]
        with Client(loop=loop, processes=False, dashboard_address=":0", scheduler_port=in_use_port) as c:
            pass
```
I get a `RuntimeError('Cluster failed to start: Scheduler failed to start.')` and not an `OSError`

https://github.com/dask/distributed/blob/5feb17151cdf660a3443abf8596444a9f51dc575/distributed/core.py#L317-L326

and even backing this out it results in: `TypeError: distributed.deploy.local.LocalCluster() got multiple values for keyword argument 'scheduler_port'`